src/components/App/ImportStudyList.js

```diff
@@ -0,0 +1,122 @@
+import React, { PureComponent } from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    TextField,
+} from '@material-ui/core';
+import { queryWebsoc } from '../../helpers';
+import store from '../../stores/RightPaneStore';
+import { addCourse, clearSchedules, openSnackbar } from '../../actions/AppStoreActions';
+import AppStore from '../../stores/AppStore';
+import { GetApp } from '@material-ui/icons';
+
+class ImportStudyList extends PureComponent {
+    state = {
+        isOpen: false,
+        selectedTerm: null,
+        studyListText: '',
+    };
+
+    handleError = (index) => {
+        clearSchedules([index]);
+        openSnackbar('error', 'An error occurred while trying to import the Study List.');
+    };
+
+    handleOpen = () => {
+        this.setState({ isOpen: true, selectedTerm: store.getFormData().term });
+    };
+
+    handleClose = (doImport) => {
```

Comment:
Would be good to make this function async to avoid .then etc.

Reply:
It'd just be a simple refactor,  so you'd do await on the Promise.all instead of chaining it with .then. I think we should use just promote using async instead of promise chaining because it looks cleaner, functionally it's identical.

Reply:
Gotcha.

Yeah, async-await definitely looks cleaner now that I've figured out how it works. Thanks for the suggestions :D